content/docs/user-guide/scripting/script-canvas/programmer-guide/custom-nodes/custom-free-function-nodes.md

```diff
@@ -318,3 +318,105 @@ Verbose usage
+
+### Step 1: Locate Generic Function Node usage
+In your Gem, you should be able to locate Generic Function Node code by searching following three macros:
+```cpp
+SCRIPT_CANVAS_GENERIC_FUNCTION_NODE(NODE_NAME, CATEGORY, UUID, DESCRIPTION, ...)
+SCRIPT_CANVAS_GENERIC_FUNCTION_NODE_WITH_DEFAULTS(NODE_NAME, DEFAULT_FUNC, CATEGORY, UUID, DESCRIPTION, ...)
+SCRIPT_CANVAS_GENERIC_FUNCTION_MULTI_RESULTS_NODE(NODE_NAME, CATEGORY, UUID, DESCRIPTION, ...)
+```
+
+Generic Function Node is deprecated and will not be supported in short future, for backward compatibility, we still support them so that existing Script Canvas Graph was created with Generic Function Node can still work.
+But we recommend you to prepare the migration as early as possible, as existing Graph will break once we remove Generic Function Node.
+
+{{< note >}}
+The migrate process assumes the node signature and behavior stay the same. If replacement node signature and behavior are different, you should upgrade Graph manually.
+{{< /note >}}
+
+### Step 2: Create replacement Custom Free Function Node
+For detailed instructions how to create Custom Free Function Node, you can refer to above topics.
+
+As an example, here is part of migration we have done for ```Math/Matrix3x3``` nodes:
+1. For ```SCRIPT_CANVAS_GENERIC_FUNCTION_NODE(NODE_NAME, CATEGORY, UUID, DESCRIPTION, ...)``` use case
+   ```cpp
+   AZ_INLINE Data::Vector3Type GetRow(const Data::Matrix3x3Type& source, Data::NumberType row) { ... }
+   SCRIPT_CANVAS_GENERIC_FUNCTION_NODE(GetRow, k_categoryName, "{C4E00343-3642-4B09-8CFA-2D2F1CA6D595}", "returns vector from matrix corresponding to the Row index", "Source", "Row");
+   ```
+   Replacement function and ScriptCanvasFunction.xml content
+   ```cpp
+   Data::Vector3Type GetRow(const Data::Matrix3x3Type& source, Data::NumberType row);
+   ```
+   ```xml
+   <Function Name="GetRow">
+       <Parameter Name="Source"/>
+       <Parameter Name="Row"/>
+   </Function>
+   ```
+
+1. For ```SCRIPT_CANVAS_GENERIC_FUNCTION_NODE_WITH_DEFAULTS(NODE_NAME, DEFAULT_FUNC, CATEGORY, UUID, DESCRIPTION, ...)``` use case
+   ```cpp
+   AZ_INLINE Data::BooleanType IsClose(const Data::Matrix3x3Type& lhs, const Data::Matrix3x3Type& rhs, const Data::NumberType tolerance) { ... }
+   SCRIPT_CANVAS_GENERIC_FUNCTION_NODE_WITH_DEFAULTS(IsClose, MathNodeUtilities::DefaultToleranceSIMD<2>, k_categoryName, "{020C2517-F02F-4D7E-9FE9-B6E91E0D6D3F}", "returns true if each element of both Matrix are equal within some tolerance", "A", "B", "Tolerance");
+   ```
+   Replacement function and ScriptCanvasFunction.xml content
+   ```cpp
+   Data::BooleanType IsClose(const Data::Matrix3x3Type& lhs, const Data::Matrix3x3Type& rhs, const Data::NumberType tolerance);
+   ```
+   ```xml
+   <Function Name="IsClose">
+       <Parameter Name="A"/>
+       <Parameter Name="B"/>
+       <Parameter Name="Tolerance" DefaultValue="0.01"/>
+   </Function>
```

Comment:
Per earlier suggestion, add an Out with a description for the IsClose function XML.